From e55b428d856acfcb56b24a0637545260bfc10886 Mon Sep 17 00:00:00 2001
From: Jianqun Xu <jay.xu@rock-chips.com>
Date: Fri, 6 Mar 2020 09:18:10 +0800
Subject: [PATCH] video/rockchip: mpp: rkvdec fix smatch warning

[smatch] drivers/video/rockchip/mpp/mpp_rkvdec.c:1239 rkvdec_devfreq_init() warn: passing zero to 'PTR_ERR'

Change-Id: Ie3777be55262034274beaa377c25febb36e83e64
Signed-off-by: Jianqun Xu <jay.xu@rock-chips.com>
---
 drivers/video/rockchip/mpp/mpp_rkvdec.c | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/drivers/video/rockchip/mpp/mpp_rkvdec.c b/drivers/video/rockchip/mpp/mpp_rkvdec.c
index 1e8cc465412a..ef8cab74772a 100644
--- a/drivers/video/rockchip/mpp/mpp_rkvdec.c
+++ b/drivers/video/rockchip/mpp/mpp_rkvdec.c
@@ -1236,11 +1236,10 @@ static int rkvdec_devfreq_init(struct mpp_dev *mpp)
 
 	dec->parent_devfreq = devfreq_get_devfreq_by_phandle(mpp->dev, 0);
 	if (IS_ERR_OR_NULL(dec->parent_devfreq)) {
-		ret = PTR_ERR(dec->parent_devfreq);
-		if (ret == -EPROBE_DEFER) {
+		if (PTR_ERR(dec->parent_devfreq) == -EPROBE_DEFER) {
 			dev_warn(mpp->dev, "parent devfreq is not ready, retry\n");
 
-			return ret;
+			return -EPROBE_DEFER;
 		}
 	} else {
 		dec->devfreq_nb.notifier_call = devfreq_notifier_call;
@@ -1252,11 +1251,10 @@ static int rkvdec_devfreq_init(struct mpp_dev *mpp)
 
 	dec->vdd = devm_regulator_get_optional(mpp->dev, "vcodec");
 	if (IS_ERR_OR_NULL(dec->vdd)) {
-		ret = PTR_ERR(dec->vdd);
-		if (ret == -EPROBE_DEFER) {
+		if (PTR_ERR(dec->vdd) == -EPROBE_DEFER) {
 			dev_warn(mpp->dev, "vcodec regulator not ready, retry\n");
 
-			return ret;
+			return -EPROBE_DEFER;
 		}
 		dev_warn(mpp->dev, "no regulator for vcodec\n");
 
-- 
2.35.3

